src/plugins/image-tiles/image-tiles-display.js

```diff
@@ -7,53 +7,50 @@ import { useService } from '../../components/container-context.js';
 import { sectionDisplayProps } from '../../ui/default-prop-types.js';
 import { HOVER_EFFECT, IMAGE_SOURCE_TYPE, LINK_SOURCE_TYPE } from './constants.js';
 
-function getSource(type, url, cdnRootUrl) {
-  switch (type) {
+function getTileImageUrl(tile, cdnRootUrl) {
+  const image = tile.image || {};
+  switch (image.sourceType) {
     case IMAGE_SOURCE_TYPE.external:
-      return url || null;
+      return image.sourceUrl;
     case IMAGE_SOURCE_TYPE.internal:
-      return url ? `${cdnRootUrl}/${url}` : null;
+      return image.sourceUrl ? `${cdnRootUrl}/${image.sourceUrl}` : '';
     default:
-      return null;
+      return '';
   }
 }
 
+const getTileLinkUrl = tile => {
+  const link = tile.link || {};
+  switch (link.sourceType) {
+    case LINK_SOURCE_TYPE.external:
+      return link.sourceUrl;
+    case LINK_SOURCE_TYPE.document:
+      return link.documentId ? urls.getDocUrl({ keyAndSlug: link.documentId }) : '';
+    default:
+      return '';
+  }
+};
+
 function ImageTilesDisplay({ content }) {
   const clientConfig = useService(ClientConfig);
 
-  const getTileUrl = tile => {
-    const link = tile.link || {};
-    switch (link.sourceType) {
-      case LINK_SOURCE_TYPE.external:
-        return link.sourceUrl;
-      case LINK_SOURCE_TYPE.document:
-        return urls.getDocUrl({ keyAndSlug: link.documentId });
-      default:
-        return '';
-    }
-  };
-
   const renderTile = (tile, index) => {
+    const linkUrl = getTileLinkUrl(tile);
+    const imageUrl = getTileImageUrl(tile, clientConfig.cdnRootUrl);
+
     const classes = classNames({
       'ImageTiles-tile': true,
-      'u-img-color-flip': content.hoverEffect === HOVER_EFFECT.colorizeZoom
+      'ImageTiles-tile--noLink': !linkUrl,
+      'u-img-color-flip': content.hoverEffect === HOVER_EFFECT.colorizeZoom,
+      'u-img-color-flip-disabled': content.hoverEffect === HOVER_EFFECT.colorizeZoom && !linkUrl
```

Comment:
I guess it would not be enough to set
```
'u-img-color-flip': !!linkUrl && content.hoverEffect === HOVER_EFFECT.colorizeZoom
```
and not set `'u-img-color-flip-disabled'` at all, but I don't quite see why 🤔

Reply:
Ah ok, cause only the hover has to be removed but the effect gets applied even when disabled.

Reply:
Because we still want the image to be grayscaled but not zoom on hover (so only one part of the style class). But now, as you asked, I think it might be a better idea to separate the current CSS class into two separately applicable classes (one for grayscale, one for zoom). I will do that, thanks for triggering the thought :smile:

Reply:
I followed your suggestion there, as it would have been more convoluted to separate the grayscale and the zoom.